.github/workflows/requirements/style.txt

```diff
@@ -0,0 +1,7 @@
+# style
+black<=23.9.1
+flake8<=6.1.0
+isort<=5.12.0
+mypy<=1.6.1
```

Comment:
Just a note that his improves portability, but reduces reproducibility. 

For instance, if CI is failing using e.g. Python 3.11 with pinned versions, then people can reproduce the exact failure on their machine using the same Python version and requirement file.

If instead the failure in CI is with Pytthon 3.11 and due to some new checks that are not supported on old versions of `mypy`, people trying it with e.g. Python 3.7 will see everything :heavy_check_mark: even using the CI setup.

Reply:
My personal preference would be for exact pinned versions in CI, not upper bounds. This ensures stability and also guarantees reproducibility. Note that black introduces backwards incompatible changes once per year, and mypy gets smarter every day, so you can and will see version bumps that introduce new formatting requirements. As long as we're aware of this, this is fine.